java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -227,6 +234,10 @@ public int hashCode() {
 
       boolean endOfPath = visitedAllParents(edge) || shouldTerminate(learnedConstraints);
 
+      if (endOfPath) {
+        flowBuilder.addAll(flowForNullableMethodParameters(edge.parent));
```

Comment:
I think this should be done outside of the path exploration as a separate step in `FlowComputation.run` method, because this is not dependent on the specific path and should be added to all flows.